src/code/z_actor.c

```diff
@@ -457,439 +1231,3485 @@ f32 Actor_XZDistanceToPoint(Actor* actor, Vec3f* point) {
+};
+
+void Actor_DrawDoorLock(GlobalContext* globalCtx, s32 frame, s32 type) {
+    s32 pad[2];
+    MtxF spA8;
+    s32 i;
+    f32 sin;
+    f32 cos;
+    struct_801AEDD4* entry = &D_801AEDD4[type];
+    f32 phi_f20;
+    f32 phi_f2;
+
+    phi_f20 = entry->unk_10;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    Matrix_InsertTranslation(0.0f, entry->unk_08, 500.0f, MTXMODE_APPLY);
+    Matrix_CopyCurrentState(&spA8);
+
+    sin = __sinf(entry->unk_00 - phi_f20) * -(10 - frame) * 0.1f * entry->unk_04;
+    cos = __cosf(entry->unk_00 - phi_f20) * (10 - frame) * 0.1f * entry->unk_04;
+
+    for (i = 0; i < 4; i++) {
+        Matrix_SetCurrentState(&spA8);
+        Matrix_InsertZRotation_f(phi_f20, MTXMODE_APPLY);
+        Matrix_InsertTranslation(sin, cos, 0.0f, MTXMODE_APPLY);
+        if (entry->unk_0C != 1.0f) {
+            Matrix_Scale(entry->unk_0C, entry->unk_0C, entry->unk_0C, MTXMODE_APPLY);
+        }
+
+        gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_OPA_DISP++, entry->unk_14);
+
+        if ((i % 2) != 0) {
+            phi_f2 = 2.0f * entry->unk_00;
+        } else {
+            phi_f2 = M_PI - (2.0f * entry->unk_00);
+        }
+        phi_f20 += phi_f2;
+    }
+
+    Matrix_SetCurrentState(&spA8);
+    Matrix_Scale(frame * 0.1f, frame * 0.1f, frame * 0.1f, MTXMODE_APPLY);
+
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(POLY_OPA_DISP++, entry->unk_18);
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+void func_800BCB50(GlobalContext* globalCtx, Vec3f* arg1) {
+    CollisionCheck_SpawnShieldParticlesMetal(globalCtx, arg1);
+}
+
+void Actor_SetColorFilter(Actor* actor, u16 colorFlag, u16 colorIntensityMax, u16 xluFlag, u16 duration) {
+    if ((colorFlag == 0x8000) && !(colorIntensityMax & 0x8000)) {
+        Actor_PlaySfxAtPos(actor, NA_SE_EN_LIGHT_ARROW_HIT);
+    }
+
+    actor->colorFilterParams = colorFlag | xluFlag | ((colorIntensityMax & 0xF8) << 5) | duration;
+    actor->colorFilterTimer = duration;
+}
+
+Hilite* func_800BCBF4(Vec3f* arg0, GlobalContext* globalCtx) {
+    Vec3f lightDir;
+
+    lightDir.x = globalCtx->envCtx.dirLight1.params.dir.x;
+    lightDir.y = globalCtx->envCtx.dirLight1.params.dir.y;
+    lightDir.z = globalCtx->envCtx.dirLight1.params.dir.z;
+
+    return func_800B7FE0(arg0, &globalCtx->view.eye, &lightDir, globalCtx->state.gfxCtx);
+}
+
+Hilite* func_800BCC68(Vec3f* arg0, GlobalContext* globalCtx) {
+    Vec3f lightDir;
+
+    lightDir.x = globalCtx->envCtx.dirLight1.params.dir.x;
+    lightDir.y = globalCtx->envCtx.dirLight1.params.dir.y;
+    lightDir.z = globalCtx->envCtx.dirLight1.params.dir.z;
+
+    return func_800B8018(arg0, &globalCtx->view.eye, &lightDir, globalCtx->state.gfxCtx);
+}
+
+void func_800BCCDC(Vec3s* points, s32 pathcount, Vec3f* pos1, Vec3f* pos2, s32 parm5) {
```

Comment:
```suggestion
void func_800BCCDC(Vec3s* points, s32 pathCount, Vec3f* pos1, Vec3f* pos2, s32 parm5) {
```
I think path count is two words right?